firmware/spiders/linksys.py

```diff
@@ -0,0 +1,111 @@
+from scrapy import Request, Spider
+from scrapy.http import Response
+from scrapy.loader import ItemLoader
+from typing import Generator
+import re
+
+from firmware.items import FirmwareItem
+
+
+class ClassIdentifier:
+    def __init__(self, shortcuts):
+        self.shortcuts = shortcuts
+
+
+class UnknownDeviceClassException(Exception):
+    def __init__(self, product):
```

Comment:
I'd think you can refactor this to 
```python
class UnknownDeviceClassException(Exception):
    pass
```
And then pass the complete error message below instead of only passing the product